cfme/middleware/providers.py

```diff
@@ -0,0 +1,32 @@
+from cfme.fixtures import pytest_selenium as sel
+from cfme.web_ui.menu import nav
+from . import list_tbl
+
+
+nav.add_branch(
+    'middleware_providers',
+    {
+        'middleware_provider_detail':
+            lambda ctx: list_tbl.click_cells({'name': ctx['provider'].name})
+    }
+)
+
+
+class Providers(object):
```

Comment:
Ahh I understand, in the framework as it stands, we do something very similar. For instance, the module name ```cfme/clouds/providers.py``` does map to the navigation (albeit changed in upstream with the addition of the new "compute" root level menu). Inside that though is a ```Provider``` class which is used to do both modeling the Cloud Provider object, and navigation. You will notice some static methods and class methods in there which allow for things like navigation etc. It would probably be advantageous to go along with the same methodology. Take a look at ```cfme/infrastructure/hosts.py``` for a smaller example of a Host object that has navigation etc in it.